### PR TITLE
Add minimal versions check to travis

### PR DESCRIPTION
**Pull request by taiki-e**
Related: https://github.com/crossbeam-rs/crossbeam/issues/312, https://github.com/rust-lang/cargo/issues/5657

### CODE REVIEW

**Comment by taiki-e** [comment]
(If adding `rand` (0.6 or later) to dependencies (not `dev-dependencies`) in the future, I think [this hack](https://github.com/rust-lang-nursery/futures-rs/blob/0.3.0-alpha.13/futures-util/Cargo.toml#L36) in `futures-rs` is helpful: https://github.com/rust-lang-nursery/futures-rs/pull/1425)

File: ci/dependencies.sh [hunk 1]
```diff
@@ -11,4 +11,8 @@ cargo tree
 cargo tree --duplicate
 cargo tree --duplicate || exit 1
 
+# Check minimal versions.
+cargo update -Zminimal-versions
+cargo tree
```
Comment (stjepang) [inline ci/dependencies.sh]: If we add a full check here, then we don't need the checks in other scripts (like `crossbeam-deque.sh`, `crossbeam-epoch.sh`, and so on):

```suggestion
cargo check --all --all-features --bins --examples --tests
```

Comment (taiki-e) [reply ci/dependencies.sh]: Tests and examples are not published to crate.io as crates, so I think that they are unnecessary. And the changes included in this PR do not allow tests and examples to be compiled with minimal versions.

Comment (stjepang) [reply ci/dependencies.sh]: That makes sense, sounds good to me! 👍

**Comment by stjepang** [comment]
Looks like the tests are failing. Perhaps we need to drop `--bins` too?

**Comment by taiki-e** [comment]
Yes, in addition, it seems necessary to exclude the benchmarks (crossbeam-channel/benchmarks).

**Comment by taiki-e** [comment]
AFAIK benchmarks are not published, but is it OK to add "publish = false" field to `Cargo.toml` for clarity?

**Comment by stjepang** [comment]
Yes, `publish = false` is a good idea since we don't want to publish that crate accidentally.

**Pull request merged**